### DISCUSSION TITLE
Deserializing only first value of each property into Properties dictionary

### DISCUSSION THREAD

**Discussion by jadurda** [Gremlinq/ExRam.Gremlinq#122] [Q&A] [answered]
Is it possible for the` Dictionary<string,object> Properties` property on an `IVertex `to return a single value cast as an object rather than a single element object array typed as an object? This would function similar to the underlying Gremlin.Net ElementMap rather than ValueMap which returns just the first property value.

**Comment by danielcweber** [comment, marked as answer]
As long as raw GraphSON is deserialized into a generic Dictionary<string,object>, you'll currently get the whole array, even if it's a singleton array. This might be configurable in the future.

**Reply by jadurda** [reply]
Thanks for the quick response and for Gremlinq. It's a significant improvement over Gremlin.Net even for my case of mostly user supplied schema-less properties which result in the dictionary of singleton arrays.
